samples/peripheral/802154_phy_test/src/comm_proc.c

```diff
@@ -0,0 +1,190 @@
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/* Purpose: Communication interface for library */
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "uart_proc.h"
+#ifdef USE_USB
+#include "usb_proc.h"
+#endif
+#include "comm_proc.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(comm);
+
+#include "ptt_uart_api.h"
+
+static int32_t comm_send_cb(const uint8_t *p_pkt, ptt_pkt_len_t len, bool add_crlf);
+
+static void comm_symbols_process(struct text_proc_s *p_text_proc, const uint8_t *buf, uint32_t len);
+static inline void text_process(struct text_proc_s *p_text_proc);
+static inline void input_state_reset(struct text_proc_s *text_proc);
+
+void comm_input_process(struct text_proc_s *p_text_proc, const uint8_t *buf, uint32_t len)
+{
+	switch (p_text_proc->state) {
+	case INPUT_STATE_IDLE:
+	if (p_text_proc->len != 0) {
+		LOG_WRN("p_text_proc->len is not 0 when input state is idle");
+		p_text_proc->len = 0;
+	}
+
+		/* start the timer after first received symbol */
+		k_timer_start(&p_text_proc->timer, K_MSEC(COMM_PER_COMMAND_TIMEOUT), K_MSEC(0));
+
+		p_text_proc->state = INPUT_STATE_WAITING_FOR_NEWLINE;
+
+		comm_symbols_process(p_text_proc, buf, len);
+	break;
```

Comment:
`break` should be at the same level as the code.

Reply:
Fixed